### ISSUE TITLE
StringIndexOutOfBoundsException in Grammar Check

### ISSUE THREAD

**Issue by ahus1** [asciidoctor/asciidoctor-intellij-plugin#718]
Reported by Sentry for plugin version 0.32.33 on IU 2021.1

```
java.lang.StringIndexOutOfBoundsException: start 1401, end 1396, length 1396
    at java.lang.AbstractStringBuilder.checkRangeSIOOBE(AbstractStringBuilder.java:1724)
    at java.lang.AbstractStringBuilder.delete(AbstractStringBuilder.java:863)
    at java.lang.StringBuilder.delete(StringBuilder.java:284)
    at com.intellij.grazie.grammar.GrammarChecker.determineTextShifts(GrammarChecker.kt:73)
    at com.intellij.grazie.grammar.GrammarChecker.check(GrammarChecker.kt:93)
    at com.intellij.grazie.ide.inspection.grammar.GrazieInspection$buildVisitor$1.visitElement(GrazieInspection.kt:102)
    at com.intellij.psi.impl.PsiElementBase.accept(PsiElementBase.java:273)
    at org.asciidoc.intellij.psi.AsciiDocSectionImpl.accept(AsciiDocSectionImpl.java:160)
    at com.intellij.codeInspection.InspectionEngine.acceptElements(InspectionEngine.java:65)
    at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.lambda$visitRestElementsAndCleanup$7(LocalInspectionsPass.java:346)
    at com.intellij.util.AstLoadingFilter.lambda$toComputable$2(AstLoadingFilter.java:174)
    at com.intellij.util.AstLoadingFilter.disallowTreeLoading(AstLoadingFilter.java:132)
    at com.intellij.util.AstLoadingFilter.disallowTreeLoading(AstLoadingFilter.java:121)
    at com.intellij.util.AstLoadingFilter.disallowTreeLoading(AstLoadingFilter.java:111)
    at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.lambda$visitRestElementsAndCleanup$10(LocalInspectionsPass.java:346)
    at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:136)
    at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:149)
    at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1096)
    at com.intellij.concurrency.ApplierCompleter.lambda$wrapInReadActionAndIndicator$1(ApplierCompleter.java:92)
    at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:688)
    at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:634)
    at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:64)
    at com.intellij.concurrency.ApplierCompleter.wrapInReadActionAndIndicator(ApplierCompleter.java:104)
    at com.intellij.concurrency.ApplierCompleter.lambda$compute$0(ApplierCompleter.java:83)
    at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:167)
    at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:178)
    at com.intellij.concurrency.ApplierCompleter.compute(ApplierCompleter.java:83)
    at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)
    at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
    at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
    at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
    at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
    at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

**Comment by ahus1** [comment]
Should be fixed in release 0.32.34

**Comment by ahus1** [comment]
Reopened, issue persists in 0.32.34. It occurs in a rare situation, but I wasn't able to reproduce it yet.

**Comment by ahus1** [comment]
Will be fixed in 0.32.35, found a way to reproduce it.

**Comment by ahus1** [comment]
Reworking it again, as it was reporting too many exceptions, even a ProcessCanceledException that is "normal" when IntelliJ notifies a background task to terminate because contents in the editor are about to change.

Will be fixed in  0.32.36
